packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedbackForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import FieldInput from 'shared/components/FieldInput';
+import { requiredField } from 'shared/components/Validation/rules';
+import { FieldTextArea } from 'shared/components/FieldTextArea';
+import { Flex, Text } from 'design';
+import Toggle from 'teleport/components/Toggle';
+import { ShareFeedbackFormValues } from './types';
+
+interface ShareFeedbackFormProps {
+  formValues: ShareFeedbackFormValues;
+
+  setFormValues(values: ShareFeedbackFormValues): void;
+}
+
+export function ShareFeedbackForm({
+  formValues,
+  setFormValues,
+}: ShareFeedbackFormProps) {
+  const isFieldRequiringEmailEnabled =
+    formValues.newsletterEnabled || formValues.salesContactEnabled;
+
+  function updateFormField<T extends keyof ShareFeedbackFormValues>(
+    field: T,
+    value: ShareFeedbackFormValues[T]
+  ) {
+    setFormValues({ ...formValues, [field]: value });
+  }
+
+  return (
+    <>
+      <FieldInput
+        mt={3}
+        type="email"
+        label="Email Address"
+        autoFocus
+        css={`
+          input {
+            font-size: 14px;
+          }
+        `}
+        rule={value =>
+          isFieldRequiringEmailEnabled
+            ? requiredField('Email is required')(value)
+            : () => ({ valid: true })
+        }
+        value={formValues.email}
+        onChange={e => updateFormField('email', e.target.value)}
+      />
+      <FieldTextArea
+        mt={1}
+        label="Any suggestions?"
+        textAreaCss={`
+                font-size: 14px;
+              `}
+        value={formValues.feedback}
+        onChange={e => updateFormField('feedback', e.target.value)}
+        placeholder="Type your suggestions here"
+      />
+      <Flex>
+        <Toggle
+          isToggled={formValues.newsletterEnabled}
+          onToggle={() => {
+            updateFormField('newsletterEnabled', !formValues.newsletterEnabled);
+          }}
+        />
+        <Text ml={2} color="text.primary">
+          Sign me up for the newsletter
+        </Text>
```

Comment:
Let's make this a label for the checkbox so that it's possible to click on the label to toggle the checkbox please.

I experimented with it a bit and it looks like we could modify `<Toggle>` to accept a child and then render the child within the label.

This works and doesn't change the visual look of the UI:

```html
<label class="Toggle__StyledWrapper-kQsuFA ekHuMv">
  <input type="checkbox" class="Toggle__StyledInput-jBswXp eetIWj">
  <div class="Toggle__StyledSlider-iAHUMy gedMTV"></div>
  <div color="text.primary" class="Text-pXEYb OoHRQ">Sign me up for the newsletter</div>
</label>
```

Reply:
Done